pkg/util/validation/validation.go

```diff
@@ -26,6 +26,22 @@ func ValidatePropagationSpec(spec policyv1alpha1.PropagationSpec) field.ErrorLis
 		allErrs = append(allErrs, field.Invalid(field.NewPath("spec").Child("propagateDeps"), spec.PropagateDeps, "application failover is set, propagateDeps must be true"))
 	}
 	allErrs = append(allErrs, ValidateFailover(spec.Failover, field.NewPath("spec").Child("failover"))...)
+	allErrs = append(allErrs, validateResourceSelectorsIfPreemptionEnabled(spec, field.NewPath("spec").Child("resourceSelectors"))...)
+	return allErrs
+}
+
+// validateResourceSelectorsIfPreemptionEnabled validates ResourceSelectors if Preemption is Always.
+func validateResourceSelectorsIfPreemptionEnabled(spec policyv1alpha1.PropagationSpec, fldPath *field.Path) field.ErrorList {
+	if spec.Preemption != policyv1alpha1.PreemptAlways {
+		return nil
+	}
+
+	var allErrs field.ErrorList
+	for index, resourceSelector := range spec.ResourceSelectors {
+		if len(resourceSelector.Name) == 0 {
```

Comment:
Maybe we can validate the `namespace` as well. 
Given all `PropagationPolicy` will have the namespace.
https://github.com/karmada-io/karmada/blob/6ef427a98e4c669c6834f0e62a400876c4b7a1c3/pkg/webhook/propagationpolicy/mutating.go#L47-L56

Reply:
Oh, no, please ignore me. 
We can't do that because for `ClusterPropagationPolicy`, the namespace might be empty even in the case of `preemption: Always`.